src/app/components/mainContent/mainContent.component.tsx

```diff
@@ -16,48 +13,34 @@ export function MainContent(): JSX.Element {
     if (content.pending) {
         return <LoadingIndicator />;
     }
-    if (content?.articles?.length) {
+    if (content.articles.length) {
         return (
             content && (
                 <div className="main-content-container">
-                    <div className="articles-container">
-                        {content?.articles.length >= 3 &&
-                            content.articles
-                                ?.slice(0, 3)
-                                .map((article: ContentItem) => (
-                                    <ContentTile
-                                        contentItem={article}
-                                        contentLabel={ContentLabel.Article}
-                                    />
-                                ))}
+                    <div className="category-items">
+                        <CategoryItems
+                            categoryItems={content.articles?.slice(0, 2)}
+                            label={ContentLabel.Article}
+                            specifiedLength={120}
+                        />
                     </div>
-
-                    <div className="reports-container">
-                        {content?.reports?.length >= 3 &&
-                            content.reports
-                                ?.slice(0, 3)
-                                .map((report: ContentItem) => (
-                                    <ContentTile
-                                        contentItem={report}
-                                        contentLabel={ContentLabel.Report}
-                                    />
-                                ))}
+                    <div className="category-items">
+                        <CategoryItems
+                            categoryItems={content.reports?.slice(0, 2)}
+                            label={ContentLabel.Report}
+                            specifiedLength={120}
+                        />
                     </div>
-
-                    <div className="blogs-container">
-                        {content?.blogs?.length >= 3 &&
-                            content.blogs
-                                ?.slice(0, 3)
-                                .map((blog: ContentItem) => (
-                                    <ContentTile
-                                        contentItem={blog}
-                                        contentLabel={ContentLabel.Blog}
-                                    />
-                                ))}
+                    <div className="category-items">
+                        <CategoryItems
+                            categoryItems={content.blogs?.slice(0, 2)}
+                            label={ContentLabel.Blog}
+                            specifiedLength={120}
+                        />
                     </div>
                 </div>
             )
         );
     }
-    return <div>test</div>;
+    return <div></div>;
```

Comment:
Do not return empty div. Refactor code above to differ between content is pending -> display loading indicator, content is available -> display content, content has error -> display error message

Reply:
refactored